sos/report/plugins/kubernetes.py

```diff
@@ -247,62 +264,76 @@ def postproc(self):
         # output that is not hit by the previous iteration.
         self.do_cmd_private_sub(self.kube_cmd)
 
+        pathexp_config_files = fr'^({"|".join(self.config_files)})'
+        self.do_file_private_sub(pathexp_config_files)
+
+        # clear base64 encoded PEM from kubeconfigs files
+        regexp = r'LS0tLS1CRUdJ[A-Za-z0-9+/=]+'
+        subst = '***** SCRUBBED BASE64 PEM *****'
+
+        self.do_path_regex_sub(pathexp_config_files, regexp, subst)
+
+        pathexp_files = fr'^({"|".join(self.files)})'
+        self.do_path_regex_sub(pathexp_files, regexp, subst)
+
 
 class RedHatKubernetes(Kubernetes, RedHatPlugin):
 
-    # OpenShift Container Platform uses the atomic-openshift-master package
-    # to provide kubernetes
-    packages = ('kubernetes', 'kubernetes-master', 'atomic-openshift-master')
+    packages = KUBE_PACKAGES + (
+        'kubernetes-master',
+        'atomic-openshift-master'
+    )
 
-    files = (
+    files = KUBECONFIGS + (
         '/etc/origin/master/admin.kubeconfig',
         '/etc/origin/node/pods/master-config.yaml',
     )
 
-    kube_cmd = "kubectl"
+    services = KUBE_SVCS
+
+    def check_enabled(self):
+        if self.is_installed("openshift-hyperkube"):
+            return False
+        return super().check_enabled()
 
     def setup(self):
-        # Rather than loading the config file, use the OCP command directly
-        # that wraps kubectl, so we don't have to manually account for any
-        # other changes the `oc` binary may implement
-        if self.path_exists('/etc/origin/master/admin.kubeconfig'):
+        self.set_kubeconfig()
+
+        # if present, use `oc` command and add some OCP specific ressources
+        if is_executable('oc'):
```

Comment:
What if `set_cubectl` sets `self.kube_cmd = "kubectl --kubeconfig={_kconf}"` and now we reset that option? Is it expected (i.e. `oc` does not honour `--kubeconfig`) or shouldnt we call `self.set_kubeconfig()` after this if-block?

Reply:
it's on purpose, I don't want to break the old behavior for old openshift installs.
as I don't have any openshift installs to look at how files are organized I just play it safe here